packages/office-ui-fabric-react/src/components/DetailsList/DetailsFooter.types.ts

```diff
@@ -0,0 +1,34 @@
+import { IRenderFunction, IStyleFunctionOrObject } from '../../Utilities';
+import { ITheme, IStyle } from '../../Styling';
+import { DetailsFooterBase } from './DetailsFooter.base';
+import { IDetailsRowProps } from './DetailsRow';
+export interface IDetailsFooter {
+  focus: () => boolean;
+}
+
+export interface IDetailsFooterProps extends React.Props<DetailsFooterBase> {
+  /**
+   * Theme provided by the Higher Order Component
+   */
+  theme?: ITheme;
+
+  /**
+   * Style function to be passed in to override the themed or default styles
+   */
+  styles?: IStyleFunctionOrObject<IDetailsFooterStyleProps, IDetailsFooterStyles>;
+  componentRef?: (component: IDetailsFooter | null) => void;
+  /**
+   * Custom classname
+   */
+  className?: string;
+  footerText?: string;
```

Comment:
Just send `columns: IColumn[]` and remove `footerText`, `footerRowProps`, and `onRenderRowFooter`. `DetailsList` has no need to customize these; your custom implementation can do these all itself.